Source/CesiumRuntime/Public/CesiumTile.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "CoreMinimal.h"
+#include "UObject/NoExportTypes.h"
+#include <Cesium3DTilesSelection/Tile.h>
+#include "CesiumTile.generated.h"
+
+/**
+ * A UObject representation of a Cesium Tile.
+ * This class provides an interface for accessing properties of a Cesium Tile
+ * from within Unreal Engine. It exposes the Bounds property, which can be
+ * accessed from Blueprints, and provides a helper function for testing
+ * intersection with a primitive component.
+ */
+UCLASS()
+class CESIUMRUNTIME_API UCesiumTile : public UObject {
+  GENERATED_BODY()
+
+  // The underlying Cesium Tile
+  const Cesium3DTilesSelection::Tile* _pTile;
+
+  // The transform of the tile
+  glm::dmat4 _transform;
+
+public:
+  /**
+   * Returns the bounds of the tile in Unreal Absolute World coordinates.
+   * This function returns the bounding box and bounding sphere that enclose the
+   * tile.
+   */
+  UFUNCTION(BlueprintCallable, Category = "Cesium")
+  FBoxSphereBounds GetBounds() const;
+
+  /**
+   * Tests whether a primitive component intersects with this tile.
+   * This function provides a convenient way to test for intersection between a
+   * primitive component and this tile.
+   */
+  UFUNCTION(BlueprintCallable, Category = "Cesium")
+  bool IsIntersecting(const UPrimitiveComponent* Other) const;
```

Comment:
I still think `IsIntersecting` and `IsContainedBy` are misleading. They imply they're testing the tile against the primitive, but really they're testing a very rough AABB or sphere approximation of the tile against the primitive. Improving the documentation to make this very clear would help, but making it clear from the names of these functions would be even better.

Here's what I'm thinking...

There's already a `GetBounds` function, and it is pretty clear about what it does. So let's make these two functions reference that.

For `IsIntersecting`:

* [ ] Rename it to `TileBoundsOverlapsPrimitive`.
* [ ] Its implementation can actually use GetBounds rather than computing it separately. The trick is to notice that there is an actual overlap only if _both_ the sphere and box overlap with the primitive.
* [ ] Update the docs to make clear that we're testing the tile's bounding box and bounding sphere against the primitive using `OverlapComponent`.

For `IsContainedBy`:

* [ ] Rename it to `PrimitiveBoxFullyContainsTileBounds`.
* [ ] Again we can implement it in terms of GetBounds, but here the tile is fully contained if _either_ the box or sphere is fully contained in the primitive's box.
* [ ] Update the docs to make clear that we're testing the tile's bounding box and bounding sphere against the primitive's axis-aligned bounding box.

This is a lot better because we're at least being clear - both in the documentation and in the naming - about exactly what is being tested. But the `PrimitiveBoxFullyContainsTileBounds` function is still a little strange. We'd really like a function that proves the tile bounds is fully inside the primitive itself, not an AABB around the primitive (which could be a lot bigger and therefore include tiles that aren't really inside the primitive at all). Is there a better way to implement it? I couldn't find one with a quick look around, unfortunately.